app/models/metric/capture.rb

```diff
@@ -128,32 +127,40 @@ def self.perf_capture_health_check(zone)
   end
   private_class_method :perf_capture_health_check
 
+  # Collect realtime targets and group them by their rollup parent
+  #
+  # 1. Only calculate rollups for Hosts
+  # 2. Some Hosts have an EmsCluster as a parent, others have none.
+  # 3. Only Hosts with a parent are rolled up.
+  #
+  # @returns Hash<String,Array<Host>>
+  #   e.g.: {"EmsCluster:4"=>[Host:4], "EmsCluster:5"=>[Host:1, Host:2]}
   def self.calc_targets_by_rollup_parent(targets)
-    # Collect realtime targets and group them by their rollup parent, e.g. {"EmsCluster:4"=>[Host:4], "EmsCluster:5"=>[Host:1, Host:2]}
-    targets_by_rollup_parent = targets.inject({}) do |h, target|
-      next(h) unless target.kind_of?(Host) && perf_capture_now?(target)
-
-      interval_name = perf_target_to_interval_name(target)
-      next unless interval_name == "realtime"
-
-      target.perf_rollup_parents(interval_name).to_a.compact.each do |parent|
+    realtime_targets = targets.select do |target|
+      target.kind_of?(Host) &&
+      perf_target_to_interval_name(target) == "realtime" &&
+      perf_capture_now?(target)
+    end
+    realtime_targets.each_with_object({}) do |target, h|
+      target.perf_rollup_parents("realtime").to_a.compact.each do |parent|
         pkey = "#{parent.class}:#{parent.id}"
-        h[pkey] ||= []
-        h[pkey] << "#{target.class}:#{target.id}"
+        (h[pkey] ||= []) << target
       end
-
-      h
     end
-    targets_by_rollup_parent
   end
   private_class_method :calc_targets_by_rollup_parent
 
-  def self.calc_tasks_by_rollup_parent(targets_by_rollup_parent)
+  # Determine queue options for each target
+  # Is only generating options for Vmware Hosts, which have a task for rollups.
+  # The rest just set the zone
+  def self.calc_target_options(zone, targets_by_rollup_parent)
     task_end_time           = Time.now.utc.iso8601
     default_task_start_time = 1.hour.ago.utc.iso8601
 
+    target_options = Hash.new { |h, k| h[k] = {:zone => zone} }
     # Create a new task for each rollup parent
-    tasks_by_rollup_parent = targets_by_rollup_parent.keys.inject({}) do |h, pkey|
+    # mark each target with the rollup parent
+    targets_by_rollup_parent.keys.each_with_object(target_options) do |pkey, h|
```

Comment:
By doing `.keys.each_with_object` here, you force yourself to have to re-lookup the value on lines 178 and 184 below.    Can you just do a normal `.each_with_object(target_options) do |(pkey, targets), h|` here?